src/header.rs

```diff
@@ -337,6 +337,11 @@ impl Header {
         bytes2path(self.path_bytes())
     }
 
+    /// Access the name portion of this header as a truncated slice of bytes.
+    pub fn name_bytes(&self) -> &[u8] {
```

Comment:
Er sorry I think I misremembered, but does `path_bytes` below work for you?

Reply:
Maybe, I'm not sure this applies to ustar paths as well. How do you think that should be handled?

Reply:
Could the conditional logic to look for the trailing slash only apply to non-ustar headers maybe? (or only one particular kind where this is known to come up?)

Reply:
What about something like this?

```
pub fn old_name_bytes(&self) -> Option<&[u8]> {
    if self.is_ustar() {
        None
    } else {
        Some(truncate(&self.as_old().name))
    }
}
```